src/test/test-sysctl-util.c

```diff
@@ -0,0 +1,43 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include "strv.h"
+#include "sysctl-util.h"
+#include "tests.h"
+
+static const char* cases[] = {
+        "a.b.c", "a/b/c",
+        "a/b/c", "a/b/c",
+        "a/b.c/d", "a/b.c/d",
+        "a.b/c.d", "a/b.c/d",
+
+        "net.ipv4.conf.enp3s0/200.forwarding", "net/ipv4/conf/enp3s0.200/forwarding",
+        "net/ipv4/conf/enp3s0.200/forwarding", "net/ipv4/conf/enp3s0.200/forwarding",
+
+        "a...b...c", "a/b/c",
+        "a///b///c", "a/b/c",
+        ".a...b...c", "a/b/c",
+        "/a///b///c", "a/b/c",
+};
+
+static void test_sysctl_normalize(void) {
+        log_info("/* %s */", __func__);
+
+        const char **s, **expected;
+        STRV_FOREACH_PAIR(s, expected, cases) {
```

Comment:
```
=================================================================
==37996==ERROR: AddressSanitizer: global-buffer-overflow on address 0x558bffb14120 at pc 0x558bffb1066c bp 0x7ffeecc64eb0 sp 0x7ffeecc64ea8
READ of size 8 at 0x558bffb14120 thread T0
    #0 0x558bffb1066b in test_sysctl_normalize /build/build/../src/test/test-sysctl-util.c:26:9
    #1 0x558bffb102e4 in main /build/build/../src/test/test-sysctl-util.c:40:9
    #2 0x7f405ebe7152 in __libc_start_main (/lib64/libc.so.6+0x27152)
    #3 0x558bffb101ed in _start (/build/build/test-sysctl-util+0x21ed)

0x558bffb14120 is located 0 bytes to the right of global variable 'cases' defined in '../src/test/test-sysctl-util.c:7:20' (0x558bffb14080) of size 160
SUMMARY: AddressSanitizer: global-buffer-overflow /build/build/../src/test/test-sysctl-util.c:26:9 in test_sysctl_normalize
Shadow bytes around the buggy address:
  0x0ab1fff5a7d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a7e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a7f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a800: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a810: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0ab1fff5a820: 00 00 00 00[f9]f9 f9 f9 00 00 00 00 00 00 00 00
  0x0ab1fff5a830: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a840: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a850: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a860: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0ab1fff5a870: 00 00 00 00 00 00 00 00 00 00 f9 f9 f9 f9 f9 f9
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
  Shadow gap:              cc
==37996==ABORTING
```

Reply:
Missing terminator. Thanks for the analysis.